lib/src/client.dart

```diff
@@ -11,22 +11,50 @@ import 'transport/abstract_transport.dart';
 import 'message/uri_pattern.dart';
 import 'protocol/session.dart';
 
+export 'package:logging/logging.dart' show Level;
+
+enum _ClientState {
+  /// Client is idle and not connected / connecting
```

Comment:
```suggestion
  /// Client is idle and not connected
```
As `connecting` is overlapping with the `waiting` state below.